torch/_inductor/ir.py

```diff
@@ -1948,12 +1949,14 @@ def _normalize_size(x, new_size):
             elif old_size[i] is None or old_size[i] == 1:
                 pass
             else:
-                # NB: new_size[i] == old_size[i] is known because the meta
-                # formula was expected to have taught us this equality.
-                # We can't conveniently check it right now because
-                # statically_known_equals doesn't know to consult preexisting
-                # guards
-                pass
+                # Sanity check: Expect broadcast compatibility
+                #
+                # NB: new_size[i] == old_size[i] is expected to already be
+                # guarded because the meta formula was expected to have taught
+                # us this equality.
+                assert (
+                    sizevars.size_hint(new_size[i] - old_size[i], fallback=0) == 0
```

Comment:
why do we need `fallback=0`?

Reply:
In the case of unbacked symints where we can't resolve the hint to an integer, it will fallback to 0 instead.

Reply:
don't we put a runtime assert in that case? the solver also checks runtime asserts

Reply:
Well @ezyang will know more but apparently lowering time is too late to generate a runtime assert

Reply:
In #124864 it says 
>  want to generate runtime assert nodes during lowering, which means that I need a finalized list of asserts by the time I start lowering.